crates/redis/src/lib.rs

```diff
@@ -87,16 +141,19 @@ impl TriggerExecutor for RedisTrigger {
         println!("Active Channels on {address}:");
         // Subscribe to channels
         for (channel, component) in self.channel_components.iter() {
-            tracing::info!("Subscribing component {component:?} to channel {channel:?}");
-            pubsub.subscribe(channel).await?;
-            println!("\t{channel}: [{}]", component.join(","));
+            if channel.contains(&address) {
+                let channel_name = channel.strip_prefix(&format!("{address}:"));
```

Comment:
I'm sorry but this stringing feels worse and worse.  Please use a tuple or struct to represent structured data.